packages/driver/src/codecs/codecs.ts

```diff
@@ -49,8 +50,94 @@ import {
 } from "./pgvector";
 import { InternalClientError } from "../errors";
 
+import type { CodecContext } from "./context";
+
 import { INVALID_CODEC_ID, KNOWN_TYPENAMES, NULL_CODEC_ID } from "./consts";
 
+import type { Float16Array } from "../adapter.shared.node";
+
+// Types for Client.withCodecs() API:
+export namespace Codecs {
+  export type Codec<T> = {
+    encode: (data: any) => T;
+    decode: (data: T) => any;
+  };
+
+  export type AnyCodec = Codec<any>;
+
+  export type BoolCodec = Codec<boolean>;
+  export type Int16Codec = Codec<number>;
+  export type Int32Codec = Codec<number>;
+  export type Int64Codec = Codec<bigint>;
+  export type Float32Codec = Codec<number>;
+  export type Float64Codec = Codec<number>;
+  export type BigIntCodec = Codec<string>;
+  export type DecimalCodec = Codec<string>;
+  export type BytesCodec = Codec<Uint8Array>;
+  export type DateTimeCodec = Codec<bigint>;
+  export type LocalDateTimeCodec = Codec<bigint>;
+  export type LocalDateCodec = Codec<
+    [years: number, months: number, days: number]
+  >;
+  export type LocalTimeCodec = Codec<bigint>;
+  export type DurationCodec = Codec<bigint>;
+  export type RelativeDurationCodec = Codec<
+    [months: number, days: number, uSeconds: bigint]
+  >;
+  export type DateDurationCodec = Codec<[months: number, days: number]>;
+  export type JsonCodec = Codec<string>;
+  export type MemoryCodec = Codec<bigint>;
+  export type PgVectorCodec = Codec<Float32Array>;
+  export type PGVectorSparseCodec = Codec<
+    [dimensions: number, indexes: Uint32Array, values: Float32Array]
+  >;
+  export type StrCodec = Codec<string>;
+  export type UUIDCodec = Codec<Uint8Array>;
+
+  // TODO: Figure out if we can drop the dep
+  // on an external package for Float16Array.
```

Comment:
Unfortunately not: only Deno, Firefox, and Safari support native Float16Array

Reply:
Sure, you mean basically vendor the existing library? We can also just use `Float32Array` until there is better support for `Float16Array`.

Reply:
I've looked at that implementation, it's huge and complicated :/ Vendoring is an option